### PR TITLE
Fix JDBC errors when addinng monitors while adding nodes

### PR DESCRIPTION
**Pull request by mklkun**

### CODE REVIEW

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java [hunk 1]
```diff
@@ -69,8 +68,7 @@ public class MonitoringService {
      * @param emsDeploymentDefinition An EMS deployment definition
      * @return return 0 if the deployment task is properly added.
      */
-    @Transactional
-    public synchronized Integer addEmsDeployment(String sessionId, EmsDeploymentDefinition emsDeploymentDefinition)
+    public Integer addEmsDeployment(String sessionId, EmsDeploymentDefinition emsDeploymentDefinition)
```
Comment (AlexFalessi) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java:71]: I believe you should keep the @Transactionnal as this method updates the DB in a loop

**Review by alijawadfahs — APPROVED** [review]

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java [hunk 2]
```diff
@@ -82,10 +80,14 @@ public synchronized Integer addEmsDeployment(String sessionId, EmsDeploymentDefi
...
-                               .forEach(node -> addEmsDeploymentForNode(node,
-                                                                        emsDeploymentDefinition.getAuthorizationBearer(),
-                                                                        emsDeploymentDefinition.isPrivateIP()));
+        emsDeploymentDefinition.getNodeNames().forEach(nodeName -> {
+            Deployment deployment = repositoryService.getDeployment(nodeName);
+            addEmsDeploymentForNode(deployment,
+                                    emsDeploymentDefinition.getAuthorizationBearer(),
+                                    emsDeploymentDefinition.isPrivateIP());
+            repositoryService.saveDeployment(deployment);
+            repositoryService.flush();
```
Comment (AlexFalessi) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java]: Is there a reason of why you flush all entities here ? `repositoryService.saveDeployment(deployment);` already saves and flush deployments entities.

If you do need to flush all entities, I think you can keep just the one below, after the loop and not flush at each iteration.

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java [hunk 3]
```diff
@@ -82,10 +80,14 @@ public synchronized Integer addEmsDeployment(String sessionId, EmsDeploymentDefi
...
+        });
 
         repositoryService.flush();
```
Comment (AlexFalessi) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java:91]: Keep only this one

**Review by AlexFalessi — APPROVED** [review]
Small non blocking comments

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java (hunk 1 above)
Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java:71]: Well I want to. But Transactional is making this to crush with a `JDBCConnectionException` error

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java (hunk 2 above)
Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java]: You're right. I should have missed to delete that one.

**Pull request merged**